src/core/or/policies.c

```diff
@@ -87,6 +89,9 @@ static const char *private_nets[] = {
   NULL
 };
 
+/** Time when we should clear IPv4/IPv6 failure entries. */
+#define IP_FAIL_CLEAR_TIME 60 * 60 * 24 * 7
```

Comment:
This time is far too long for mobile devices and laptops, which can change networks every few seconds. It's also far too long for typical users, who give up after 30 seconds.